### PR TITLE
Create getProposedFederationSize in Federation Support

### PR DESCRIPTION
**Pull request by apancorb**
## Description
Create getProposedFederationSize in Federation Support.

## Motivation and Context
https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP419.md

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 1]
```diff
@@ -2208,6 +2208,32 @@ void getProposedFederation_whenStorageProviderReturnsProposedFederation_shouldRe
...
+        // Assert
+        assertEquals(expectedCode, actualProposedFederationSize);
+    }
+
+    @Test
+    void getProposedFederationSize_whenStorageProviderReturnsProposedFederation_shouldReturnProposedFederationSize() {
+        // Arrange
+        Federation proposedFederation = P2shErpFederationBuilder.builder().build();
+        storageProvider.setProposedFederation(proposedFederation);
+        int expectedSize = proposedFederation.getSize();
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2229]:
```suggestion
        int expectedSize = 9;
```

same here, to make it deterministic

Comment (apancorb) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2229]: Why is this not deterministic?

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationChangeResponseCode.java [hunk 2]
```diff
@@ -8,6 +8,7 @@ public enum FederationChangeResponseCode {
     PROPOSED_FEDERATION_ALREADY_EXISTS(-4),
     PENDING_FEDERATION_NON_EXISTENT(-1),
     RETIRING_FEDERATION_NON_EXISTENT(-1),
+    PROPOSED_FEDERATION_NON_EXISTENT(-1),
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/federation/FederationChangeResponseCode.java:11]:
```suggestion
    FEDERATION_NON_EXISTENT(-1),
```
What do you think? We wrap it in a single response code

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java [hunk 3]
```diff
@@ -375,6 +375,13 @@ public Optional<Federation> getProposedFederation() {
         return provider.getProposedFederation(constants, activations);
     }
 
+    @Override
+    public int getProposedFederationSize() {
+        return getProposedFederation()
+            .map(Federation::getSize)
+            .orElse(FederationChangeResponseCode.PROPOSED_FEDERATION_NON_EXISTENT.getCode());
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java:382]: I know this is how it's handled with the retiring federation. But I wonder, should we perhaps return an Optional here and then et the Bridge decide to return `-1` when there is no proposed federation?

I feel like it's the Bridge responsibility to decide on the default value. For any other internal uses of this method, an Optional return might be more useful

Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java:382]: Seems like `getRetiringFederationCreationTime` is handled that way

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationChangeResponseCode.java (hunk 2 above)
Comment (apancorb) [reply rskj-core/src/main/java/co/rsk/peg/federation/FederationChangeResponseCode.java]: Sounds good, updated all references

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java (hunk 3 above)
Comment (apancorb) [reply rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java]: Yes, I agree. I was mainly just imitating how retiring federation does it. I have updated. Hopefully, we can do this for the rest of the methods and let the Bridge decide the defaults.

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationChangeResponseCode.java [hunk 4]
```diff
@@ -6,8 +6,7 @@ public enum FederationChangeResponseCode {
     EXISTING_FEDERATION_AWAITING_ACTIVATION(-2),
     RETIRING_FEDERATION_ALREADY_EXISTS(-3),
     PROPOSED_FEDERATION_ALREADY_EXISTS(-4),
-    PENDING_FEDERATION_NON_EXISTENT(-1),
-    RETIRING_FEDERATION_NON_EXISTENT(-1),
+    FEDERATION_NON_EXISTENT(-1),
```
Comment (julia-zack) [inline rskj-core/src/main/java/co/rsk/peg/federation/FederationChangeResponseCode.java:9]: mmm i won't do this, we should aim to have different codes in the future, not to unify them

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 5]
```diff
@@ -2208,6 +2208,32 @@ void getProposedFederation_whenStorageProviderReturnsProposedFederation_shouldRe
...
+    }
+
+    @Test
+    void getProposedFederationSize_whenStorageProviderReturnsProposedFederation_shouldReturnProposedFederationSize() {
+        // Arrange
+        List<BtcECKey> federationKeys = BitcoinTestUtils.getBtcEcKeysFromSeeds(
+            new String[] { "fa01", "fa02", "fa03", "fa04", "fa05", "fa06", "fa07", "fa08", "fa09" }, true);
+        Federation proposedFederation = P2shErpFederationBuilder.builder().withMembersBtcPublicKeys(federationKeys).build();
+        storageProvider.setProposedFederation(proposedFederation);
+        int expectedSize = proposedFederation.getSize();
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2228]:
```suggestion
        int expectedSize = federationKeys.getSize();
```

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationChangeResponseCode.java (hunk 4 above)
Comment (apancorb) [reply rskj-core/src/main/java/co/rsk/peg/federation/FederationChangeResponseCode.java:9]: https://github.com/rsksmart/rskj/pull/2782#discussion_r1784991712

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**